consensus/hotstuff/votecollector/vote_collector_statemachine.go

```diff
@@ -0,0 +1,141 @@
+package votecollector
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+)
+
+var (
+	ErrInvalidCollectorStateTransition = errors.New("invalid state transition")
+	ErrDifferentCollectorState         = errors.New("different state")
+)
+
+// NewVerifyingCollectorFactoryMethod is a factory method to generate a hotstuff.VoteCollectorState
+type NewVerifyingCollectorFactoryMethod = func(base BaseVoteCollector) (hotstuff.VoteCollectorState, error)
+
+// VoteCollectorStateMachine implements a state machine for transition between different states of vote collector
+type VoteCollectorStateMachine struct {
```

Comment:
A bit verbose since the namespace already have votecollector.
```suggestion
type StateMachine struct {
```